src/Nox.Generator/Infrastructure/Persistence/DbContext.template.cs

```diff
@@ -134,10 +135,25 @@ private void AuditEntity(EntityEntry<AuditableEntityBase> entry)
             case EntityState.Deleted:
                 entry.State = EntityState.Modified;
                 entry.Entity.Deleted(user, system);
+                ReattachCompoundTypes(entry);
                 break;
         }
     }
 
+    private void ReattachCompoundTypes(EntityEntry<AuditableEntityBase> parentEntry)
+    {
+        foreach (var navigationEntry in parentEntry.Navigations)
+        {
+            foreach (var typeEntry in ChangeTracker.Entries<INoxType>())
```

Comment:
@dejobratic  why INoxType and not IOwnedEntity? INoxType is for fields

Reply:
The issue has been observed for Nox compound types, like StreetAddress and TranslatedText, which are registered in EF as owned entities, but they are of INoxType.

Reply:
and other owned entities, for example CountryLocalNames in Countries, do we have the same issue?